src/components/CodeShower.vue

```diff
@@ -65,9 +65,9 @@ let text =
 @inputs EGP:wirelink ${this.needUserInteraction() ? "User:entity" : ""}
 @persist ActiveScreen:number
 
-if(first() | duped() | dupefinished()) {
+if(first() | ~EGP | dupefinished()) {
```

Comment:
Would lead to **redundant function declarations** everytime we connect a screen.

Reply:
> Would lead to **redundant function declarations** everytime we connect a screen.

Then move the definitions outside of the "entry point".

Reply:
Moving function definitions on top-level scope c

Reply:
We can move function definitions to the top-level scope if we decide to go from the old [trigger system](https://github.com/wiremod/wire/wiki/E2:-Triggers) to the new [event system](https://github.com/wiremod/wire/wiki/Expression-2-Events) as it does not execute the whole chip's code at each input change.
As long as we use [triggers](https://github.com/wiremod/wire/wiki/E2:-Triggers) we can't move the function definitions as they will be uselessly executed at each input change.

If you're interested in making this (small) transition, feel free to open an [issue](https://github.com/NiTrO0FuN/easy-e2/issues).

(N.B: I think that because [triggers](https://github.com/wiremod/wire/wiki/E2:-Triggers) are not fully deprecated, using [events](https://github.com/wiremod/wire/wiki/Expression-2-Events) will still not allow us to move function definitions as the chip will still be re-executed for each input change.)